### PR TITLE
Add types for Editor and Setting.

### PR DESCRIPTION
**Pull request by mtgto**
- Add types for Editor and Setting
- Add `@types/lodash`

### CODE REVIEW

File: src/lib/Util/findQueryByLine.ts [hunk 1]
```diff
@@ -10,6 +10,13 @@ export default function findQueryByLine(sql: string, line: number): QueryChunk {
   const chunks = splitQuery(sql);
   const chunk = find(chunks, chunk => chunk.endLine >= line) || last(chunks);
 
+  if (!chunk) {
```
Comment (mtgto) [inline src/lib/Util/findQueryByLine.ts]: find and last can returns undefined

File: src/renderer/components/Editor/Editor.tsx [hunk 2]
```diff
@@ -36,9 +46,9 @@ export default class Editor extends React.Component<any, any> {
       [process.platform === "darwin" ? "Cmd-/" : "Ctrl-/"]: () => {
         this.codeMirror.execCommand("toggleComment");
       },
-      Tab: cm => {
+      Tab: (cm: CodeMirror.Editor) => {
         if (!cm.state.vim) {
-          if (cm.somethingSelected()) cm.indentSelection("add");
+          if (cm.getDoc().somethingSelected()) cm.execCommand("indentMore");
```
Comment (mtgto) [inline src/renderer/components/Editor/Editor.tsx]: `indentSelection` is not public function.
https://github.com/codemirror/CodeMirror/pull/6083#issuecomment-563124716

File: src/renderer/components/Editor/Editor.tsx [hunk 3]
```diff
@@ -70,8 +80,8 @@ export default class Editor extends React.Component<any, any> {
     }
   }
 
-  componentWillReceiveProps(nextProps) {
-    if (nextProps.value !== undefined && this.currentValue !== nextProps.value) {
+  componentWillReceiveProps(nextProps: Props) {
+    if (this.currentValue !== nextProps.value) {
```
Comment (mtgto) [inline src/renderer/components/Editor/Editor.tsx]: value is always non-null.

File: src/lib/GitHubApiClient.ts [hunk 4]
```diff
@@ -1,35 +1,37 @@
 const GITHUB_COM_URL = "https://api.github.com";
 
 export default class GitHubApiClient {
-  baseUrl: string;
-  token: string;
+  readonly baseUrl: string;
+  readonly token: string | null;
 
-  constructor({ token, url }) {
-    this.baseUrl = url;
```
Comment (mtgto) [inline src/lib/GitHubApiClient.ts]: To avoid to have nullable value in member

**Comment by hokaccha** [comment]
@mtgto Please resolve the conflict 🙏

**Pull request merged**